src/GenericConnectionHandler.cpp

```diff
@@ -22,11 +22,10 @@
 static inline ConnectionHandler* instantiate_handler(NetworkAdapter adapter);
 
 bool GenericConnectionHandler::updateSetting(const models::NetworkSetting& s) {
-
-    if(_ch != nullptr && _ch->_current_net_connection_state != NetworkConnectionState::INIT) {
+    if(_ch != nullptr && _current_net_connection_state != NetworkConnectionState::INIT) {
```

Comment:
Can you provide me an explanation for using `_current_net_connection_state` instead of  `_ch->_current_net_connection_state`?

Reply:
As in description the _ch->_current_net_connection_state is never updated since the state is updated in the `check` method, BUT the `check` of `_ch ` obj is never called so the state of _ch is always "INIT" even if you are trying to connect and the state of GenericConnectionHandler is "CONNECTING"

Reply:
Instead of doing this I think it may be better to modify check, and make it virtual and make it update the state of the wrapped object. The state of the wrapper object is meaningless in my opinion.

Reply:
I believed that there was a reason for wrapping only the `update_*` method instead wrapping directly the `check`...
BTW I am updating the PR soon.